### PR TITLE
LAB01: Fibonacci Series

### PR DESCRIPTION
**Pull request by ChitwanAgrawal**
#### Changes proposed in this PR:

- Program for Printing Fibonacci Series

Fixes #24

### CODE REVIEW

File: 01_fib/main.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	num1 = 0
+	num2 = 1
+	sum  = 0
+)
+
+func fib(n int) {
+	for i := 0; i < n; i++ {
+		sum = num1 + num2
+		num1 = num2
+		num2 = sum
```
Comment (paraganz) [inline 01_fib/main.go]: As mentioned by Marcelo in my PR, this can be num1, num2 = num2, sum

File: 01_fib/main_test.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	// Given
+	r := require.New(t)
+	var buf bytes.Buffer
+	out = &buf
+
+	// When
+	main()
+
+	// Then
+	expected := strconv.Quote("1\n2\n3\n5\n8\n13\n21\n")
```
Comment (paraganz) [inline 01_fib/main_test.go]: https://github.com/anz-bank/go-training/pull/20#discussion_r255423680

Comment (paraganz) [inline 01_fib/main_test.go]: This should print 1\n1\n2\n3\n5\n8\n13. Please check the implementation.

File: 01_fib/main.go (hunk 1 above)
Comment (ChitwanAgrawal) [reply 01_fib/main.go]: Done.Thanks.

**Comment by anzdaddy** [comment]
Great to see @paraganz on the case!

**Comment by anzdaddy** [comment]
Your code isn't auto-formatted. Do you have VSCode and the Go extension installed?

File: 01_fib/main_test.go (hunk 2 above)
Comment (ChitwanAgrawal) [reply 01_fib/main_test.go]: Done. Whatever was aligned with me.

Comment (ChitwanAgrawal) [reply 01_fib/main_test.go]: Done.

**Comment by ChitwanAgrawal** [comment]
> Your code isn't auto-formatted. Do you have VSCode and the Go extension installed?

Yes I am using VSCode with Go Extension 0.9.1 installed. Do I need to install any other extension for formatting?

**Comment by anzdaddy** [comment]
> > Your code isn't auto-formatted. Do you have VSCode and the Go extension installed?
> 
> Yes I am using VSCode with Go Extension 0.9.1 installed. Do I need to install any other extension for formatting?

Shouldn't be necessary. Maybe it's misconfigured.

**Comment by juliaogris** [comment]
For auto-formatting try:
>Open Settings (Ctrl + ,)
Search for editor.formatOnSave and set to true
You Go code will be formatted automatically on Ctrl + s

(https://stackoverflow.com/a/53513867)

File: 01_fib/main.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+var (
```
Comment (juliaogris) [inline 01_fib/main.go:11]: use local variables rather than global ones.

File: 01_fib/main.go [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+	num1, num2 = 0, 1
+	sum        = 0
+)
+
+func fib(n int) {
+	fmt.Fprintln(out, num2)
```
Comment (juliaogris) [inline 01_fib/main.go:17]: One print state should suffice:
```go
	a, b := 1, 1
	for i := 1; i <= n; i++ {
		fmt.Fprintln(out, a)
		a, b = b, a+b
	}
```

File: 01_fib/main_test.go [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+	// Given
+	r := require.New(t)
+	var buf bytes.Buffer
+	out = &buf
+
+	// When
+	main()
+
+	// Then
+	expected := strconv.Quote(`1
```
Comment (juliaogris) [inline 01_fib/main_test.go:21]: Nice! I like the string literal.

File: 01_fib/main_test.go [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestMainOutput(t *testing.T) {
```
Comment (juliaogris) [inline 01_fib/main_test.go:11]: Test corner cases too: -1, 0, very big number that causes an overflow

**Review by juliaogris — COMMENTED** [review]
Please remove the binary `01_fib` from your existing commits and force push - binaries even when removed blow out the repo size as all of the history is cloned.
Please update PR description to reference issue and to render as proper markdown `- `should do the trick.
Please edit your commits with `git rebase -i HASH` and force push:
* Capitalize the subject line but use Title case (no all caps)
* No need for `LAB01: ` prefix
* Fold fixup commits into previous ones

File: 01_fib/main.go (hunk 4 above)
Comment (anz-rfc) [inline 01_fib/main.go:17]: It's better to decouple fibonacci sequence calculation from IO. https://github.com/anz-bank/go-training/issues/37

**Pull request closed without merging**